File: java/core/src/main/java/com/github/fernthedev/lightchat/core/VersionData.java
package com.github.fernthedev.lightchat.core;

import lombok.*;
import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
import org.jetbrains.annotations.Nullable;

@Getter
@RequiredArgsConstructor
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class VersionData {

    @Getter
    @Nullable
    private VariablesJSON variablesJSON;

    @NonNull
    private DefaultArtifactVersion version;

    @NonNull
    private DefaultArtifactVersion minVersion;

    public VersionData(VariablesJSON variablesJSON) {
        this(variablesJSON.getVersion(), variablesJSON.getMinVersion());
        this.variablesJSON = variablesJSON;
    }

    public VersionData(String version, String minVersion) {
        this.version = new DefaultArtifactVersion(version);
        this.minVersion = new DefaultArtifactVersion(minVersion);
    }






}

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/VersionData.kt
package com.github.fernthedev.lightchat.core

import com.github.fernthedev.lightchat.core.packets.handshake.VersionDataString
import org.apache.maven.artifact.versioning.DefaultArtifactVersion

data class VersionData
@JvmOverloads
constructor(
    val version: DefaultArtifactVersion,
    val minVersion: DefaultArtifactVersion,
    val variablesJSON: VariablesJSON? = null,
) {

    constructor(variablesJSON: VariablesJSON) : this(variablesJSON = variablesJSON, version = DefaultArtifactVersion(variablesJSON.version), minVersion = DefaultArtifactVersion(variablesJSON.minVersion))
    constructor(variablesJSON: VersionDataString) : this(variablesJSON = null, version = DefaultArtifactVersion(variablesJSON.version), minVersion = DefaultArtifactVersion(variablesJSON.minVersion))
    constructor(v: String, mv: String) : this(variablesJSON = null, version = DefaultArtifactVersion(v), minVersion = DefaultArtifactVersion(mv))

}